plugins/modules/system/listen_ports_facts.py

```diff
@@ -220,18 +299,34 @@ def getPidUser(pid):
     }
 
     try:
-        netstat_cmd = module.get_bin_path('netstat', True)
+        command = None
+        bin_path = None
+        if module.params['command'] is not None:
+            command = module.params['command']
+            bin_path = module.get_bin_path(command, required=True)
+        else:
+            for c in commands_map.keys():
+                bin_path = module.get_bin_path(c, required=False)
+                if bin_path is not None:
+                    command = c
+                    break
+
+        if bin_path is None:
+            raise EnvironmentError(msg='Unable to find any of the supported commands in PATH: {0}'.format(", ".join(commands_map.keys())))
 
         # which ports are listening for connections?
-        rc, stdout, stderr = module.run_command([netstat_cmd, '-plunt'])
+        args = commands_map[command]['args']
+        rc, stdout, stderr = module.run_command([bin_path, args])
```

Comment:
Since you are using `args` and not `arg`, you want to change `commands_map` to provide a list and not a string, and replace this with
```suggestion
        rc, stdout, stderr = module.run_command([bin_path] + args)
```